x-pack/plugins/security_solution/common/api/detection_engine/prebuilt_rules/model/diff/rule_diff/rule_diff.ts

```diff
@@ -63,6 +63,10 @@ export type RuleFieldsDiff = CommonFieldsDiff &
 export interface FullRuleDiff {
   fields: RuleFieldsDiff;
   has_conflict: boolean;
+  has_non_solvable_conflict: boolean;
+  num_fields_with_updates: number;
+  num_fields_with_conflicts: number;
+  num_fields_with_non_solvable_conflicts: number;
```

Comment:
If we have `num_fields_with_conflicts` and `num_fields_with_non_solvable_conflicts`, do we still need `has_conflict` and `has_non_solvable_conflict`?

Reply:
These two fields simplified the calculation of `num_rules_with_conflicts` and `num_rules_with_non_solvable_conflicts` in `calculateRuleStats` in `x-pack/plugins/security_solution/server/lib/detection_engine/prebuilt_rules/api/review_rule_upgrade/review_rule_upgrade_route.ts`.

But, yeah, agree that its cleaner to remove them from the types and calculate these two in another way. I rewrote the `calculateRuleStats` method.